server/src/server-plugin.ts

```diff
@@ -0,0 +1,22 @@
+import { Plugin } from '@cuillere/core'
+import { Context, ContextFunction, GraphQLServiceContext } from 'apollo-server-core'
+import { GraphQLServerListener, ValueOrPromise } from 'apollo-server-plugin-base'
+
+import { ApolloServerPluginArgs } from './apollo-server-plugin'
+import { KoaMiddlewareArgs } from './koa-middleware'
+import { TaskListener } from './task-manager'
+
+export type ServerPlugin = {
+  graphqlContext?: Context | ContextFunction
+  httpRequestListeners?: OneOrMany<GetTaskListener<KoaMiddlewareArgs>>
+  graphqlRequestListeners?: OneOrMany<GetTaskListener<ApolloServerPluginArgs>>
+  plugins?: OneOrMany<Plugin>
+  serverWillStart?: (service: GraphQLServiceContext) => ValueOrPromise<GraphQLServerListener | void>
```

Comment:
On leur permet de renvoyer eux meme des listener pour du plugin apolo ?

Reply:
This is designed to allow the postgres ServerPlugin to build the crud object when the server starts...
It will be called by the actual `serverWillStart` of the existing apollo-server-plugin.
For now I saw no good reason not to use the same API as the original `serverWillStart`.